packages/common/amplify_db_common/lib/src/connect_io.dart

```diff
@@ -12,55 +12,21 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-import 'dart:io';
-import 'dart:isolate';
-
+import 'package:amplify_db_common_dart/src/connect_io.dart' as base;
 import 'package:drift/drift.dart';
-import 'package:drift/isolate.dart';
-import 'package:drift/native.dart';
 import 'package:path/path.dart' as p;
+import 'package:path_provider/path_provider.dart';
 
 /// {@macro amplify_db_common.connect}
-///
-/// Creates the executor in a new background isolate.
-/// See [Using drift in a background isolate](https://drift.simonbinder.eu/docs/advanced-features/isolates/#initialization-on-the-main-thread)
-/// and [the drift native example](https://github.com/simolus3/drift/blob/3253cd7ead51d7b163542929eb0fc8daee573fce/examples/app/lib/database/connection/native.dart)
-/// for more info.
 QueryExecutor connect({
   required String name,
   String? path,
 }) {
-  assert(path != null, 'path cannot be null on vm.');
-
-  return DatabaseConnection.delayed(
```

Comment:
I was able to get drift working on iOS with this same code.  Is it necessary that we have the amplify_db_common and amplify_db_common_dart?  

Couldn't we just have amplify_sqlite3_flutter_libs (which would be sqlite3_flutter_libs without iOS) and then the existing amplify_db_common?

Reply:
This adds path_provider to the connect function. The separation is nice if we ever need it for other things as well. A bit of overhead but I don't think it's too bad.

We don't want to bring in DataStore as a dependency though? We just want to match the SQLite version used which does require a separate package.